src/table.js

```diff
@@ -1322,6 +1322,84 @@ Table.prototype.sampleRowKeysStream = function(gaxOptions) {
   ]);
 };
 
+/**
+ * Returns true in callback if consitency check is true for
+ * generated consistency token.
+ * In-case consistency check returns false, retrial is done in interval
+ * of 5 seconds till 10 minutes, after that it returns false.
+ *
+ * @param {object} [gaxOptions] Request configuration options, outlined here:
+ *     https://googleapis.github.io/gax-nodejs/CallSettings.html.
+ *
+ * @param {function(?error, ?boolean)} callback The callback function.
+ * @param {?Error} callback.err An error returned while making this request.
+ * @param {?Boolean} callback.resp Boolean value.
+ */
+Table.prototype.waitForReplication = function(callback) {
+  let reqOpts = {
+    name: this.id,
+  };
+
+  this.bigtable
+    .request({
+      client: 'BigtableTableAdminClient',
+      method: 'generateConsistencyToken',
+      reqOpts: reqOpts,
+    })
+    .then(resp => {
+      const cToken = resp[0].consistencyToken;
+
+      checkConsistency(cToken, (err, res) => {
+        if (!err && res === true) {
+          return callback(null, true);
+        } else {
+          let intervalCall = setInterval(
```

Comment:
I rewrote this a bit to switch from the interval to timeouts. The code below also incorporates some other suggestions, which I left comments for (please remove):

```js
Table.prototype.waitForReplication = function(callback) {
  const self = this;

  const reqOpts = {
    name: this.id,
  };

  this.bigtable
    .request({
      client: 'BigtableTableAdminClient',
      method: 'generateConsistencyToken',
      reqOpts: reqOpts,
    },
    // Switched to using a callback here, for consistency with our other methods
    (err, resp) => {
      if (err) {
        callback(err);
        return;
      }

      const timeoutAfterTenMinutes = setTimeout(() => {
        callback(null, false);
      }, 10 * 60 * 1000);

      function retryIfNecessary(err, res) {
        // Should we retry on specific errors, as opposed to all?
        // For example, if there is a network error or auth fails
        // for some reason, maybe the user needs to know?
        if (!err && res === true) {
          clearTimeout(timeoutAfterTenMinutes);
          callback(null, true);
          return;
        }

        // Using a `setTimeout` here instead of the interval. This
        // will guarantee there isn't a race condition between two
        // calls, i.e. if we send two requests, 5 seconds apart, but
        // for some reason, the first request responds after the 
        // second one.
        setTimeout(checkConsistency, 5000);
      }

      function checkConsistency() {
        const token = resp[0].consistencyToken;
        self.checkConsistency(token, retryIfNecessary);
      }
    }
  );
};

// I moved this to a new method, since it seems useful on its own.
// This will also make testing a bit easier.
Table.prototype.checkConsistency = function(token, callback) {
  const reqOpts = {
    name: this.id,
    consistencyToken: token,
  };

  this.bigtable.request(
    {
      client: 'BigtableTableAdminClient',
      method: 'checkConsistency',
      reqOpts: reqOpts,
    },
    (err, resp) => {
      if (err) {
        callback(err);
        return;
      }

      callback(null, resp.consistent === 1);
    }
  );
}

```

Reply:
perfect, 'll update this by today